src/vos/vos_pool.c

```diff
@@ -519,93 +519,27 @@ vos_pool_destroy(const char *path, uuid_t uuid)
 }
 
 static int
-set_slab_prop(int id, struct pobj_alloc_class_desc *slab)
+vos_register_slabs(struct umem_attr *uma)
 {
-	struct daos_tree_overhead	ovhd = { 0 };
-	int				tclass, *size, rc;
+	struct pobj_alloc_class_desc *slab;
+	int                           i, rc;
+	int                           defined;
+	int                           used = 0;
 
-	if (id == VOS_SLAB_OBJ_DF) {
-		slab->unit_size = sizeof(struct vos_obj_df);
-		goto done;
+	for (i = 0; i < ARRAY_SIZE(slab_map); i++) {
+		if (slab_map[i] != -1)
+			used |= (1 << i);
 	}
 
-	size = &ovhd.to_leaf_overhead.no_size;
-
-	switch (id) {
-	case VOS_SLAB_OBJ_NODE:
-		tclass = VOS_TC_OBJECT;
-		break;
-	case VOS_SLAB_KEY_NODE:
-		tclass = VOS_TC_DKEY;
-		break;
-	case VOS_SLAB_SV_NODE:
-		tclass = VOS_TC_SV;
-		break;
-	case VOS_SLAB_EVT_NODE:
-		tclass = VOS_TC_ARRAY;
-		break;
-	case VOS_SLAB_EVT_NODE_SM:
-		tclass = VOS_TC_ARRAY;
-		size = &ovhd.to_int_node_size;
-		break;
-	case VOS_SLAB_EVT_DESC:
-		tclass = VOS_TC_ARRAY;
-		size = &ovhd.to_record_msize;
-		break;
-	default:
-		D_ERROR("Invalid slab ID: %d\n", id);
-		return -DER_INVAL;
-	}
-
-	rc = vos_tree_get_overhead(0, tclass, 0, &ovhd);
-	if (rc)
-		return rc;
-
-	slab->unit_size = *size;
-done:
-	D_ASSERT(slab->unit_size > 0);
-	D_DEBUG(DB_MGMT, "Slab ID:%d, Size:%lu\n", id, slab->unit_size);
-
-	slab->alignment = 0;
-	slab->units_per_block = 1000;
-	slab->header_type = POBJ_HEADER_NONE;
-
-	return 0;
-}
-
-static int
-vos_register_slabs(struct umem_attr *uma)
-{
-	struct pobj_alloc_class_desc	*slab;
-	int				 i, rc, j;
-	bool				 skip_set;
-
 	D_ASSERT(uma->uma_pool != NULL);
-	for (i = 0; i < VOS_SLAB_MAX; i++) {
+	for (i = 0; i < UMM_SLABS_CNT; i++) {
+		D_ASSERT(used != 0);
+		defined               = __builtin_ctz(used);
 		slab = &uma->uma_slabs[i];
-
-		D_ASSERT(slab->class_id == 0);
-		rc = set_slab_prop(i, slab);
-		if (rc) {
-			D_ERROR("Failed to get unit size %d. rc:%d\n", i, rc);
-			return rc;
-		}
-
-		skip_set = false;
-		for (j = 0; j < i; j++) {
-			if (uma->uma_slabs[j].unit_size == slab->unit_size) {
-				/** PMDK will fail to register a new slab of the same size
-				 *  so reuse the class id
-				 */
-				slab->class_id = uma->uma_slabs[j].class_id;
-				skip_set = true;
-				D_ASSERT(slab->class_id != 0);
-				break;
-			}
-		}
-
-		if (skip_set)
-			continue;
+		slab->alignment       = 0;
+		slab->unit_size       = (defined + 1) * 32;
+		slab->units_per_block = 1000;
```

Comment:
what is the default if we don't set this one?

Reply:
It is a minimum specification.  If you look at my reply to Niu's comment, it looks like 256K is the default for all of the internal types.  Maybe we should follow suit or let PMDK pick.